timemachine/cpp/src/nonbonded_interaction_group.cu

```diff
@@ -41,17 +42,31 @@ NonbondedInteractionGroup<RealType>::NonbondedInteractionGroup(
                     &k_nonbonded_unified<RealType, 1, 1, 0>,
                     &k_nonbonded_unified<RealType, 1, 1, 1>}),
 
-      beta_(beta), cutoff_(cutoff), nblist_(N_), nblist_padding_(nblist_padding), d_sort_storage_(nullptr),
+      beta_(beta), cutoff_(cutoff), nblist_(NR_ + NC_), nblist_padding_(nblist_padding), d_sort_storage_(nullptr),
```

Comment:
For Local MD this change of `NR_ + NC_` will likely cause failures, as it reuse the potential with different sizes of indices, I would probably keep it `N_`. Otherwise, just need to modify https://github.com/proteneer/timemachine/blob/split_ff_solv_ixn_gpu/timemachine/cpp/src/local_md_utils.cu#L63 to construct ixn group that has size N


```suggestion
      beta_(beta), cutoff_(cutoff), nblist_(N_), nblist_padding_(nblist_padding), d_sort_storage_(nullptr),
```